libraries/lib-wave-track/WaveClip.cpp

```diff
@@ -719,8 +727,19 @@ void WaveClip::WriteXML(size_t ii, XMLWriter &xmlFile) const
 }
 
 /*! @excsafety{Strong} */
-bool WaveClip::Paste(double t0, const WaveClip& other)
-{
+bool WaveClip::Paste(double t0, const WaveClip& o)
+{
+   const WaveClip *pOther = &o;
+   WaveClipHolder dup;
+   if(!o.StrongInvariant()) {
+      assert(false); // precondition not honored
+      // But try to repair it and continue in release
+      auto dup = std::make_shared<WaveClip>(o, o.GetFactory(), true);
+      dup->RepairChannels();
+      pOther = dup.get();
```

Comment:
Looks like a bug: `dup` gets destroyed when this scope closes, making `pOther` unusable.

Reply:
Sharp eyes